Lib/fontTools/varLib/__init__.py

```diff
@@ -700,56 +676,141 @@ def load_designspace(designspace):
+
+		# Sanity check axis values to match current avar requirements.
+		if axis.map:
+			input_locations = {input for input, _ in axis.map}
+			output_locations = {output for _, output in axis.map}
+
+			# If any axis has a mapping, it must include a mapping for the axis minimum,
+			# default and maximum values:
+			if axis.minimum not in input_locations:
+				raise VarLibValidationError(f"Designspace {ds_path}: Axis '{axis.name}' has a mapping and must therefore include an entry for its own minimum value {axis.minimum} (<map input='{axis.minimum}' output='...'>).")
+			if axis.minimum != min(input_locations):
+				raise VarLibValidationError(f"Designspace {ds_path}: Axis '{axis.name}': the axis minimum value must be the lowest input value in the mapping.")
+			axis_min_output = axis.map_forward(axis.minimum)
+			if axis_min_output != min(output_locations):
+				raise VarLibValidationError(f"Designspace {ds_path}: Axis '{axis.name}': the axis minimum value must be the lowest output value in the mapping.")
+
+			if axis.default not in input_locations:
+				raise VarLibValidationError(f"Designspace {ds_path}: Axis '{axis.name}' has a mapping and must therefore include an entry for its own default value {axis.default} (<map input='{axis.default}' output='...'>).")
+
+			if axis.maximum not in input_locations:
+				raise VarLibValidationError(f"Designspace {ds_path}: Axis '{axis.name}' has a mapping and must therefore include an entry for its own maximum value {axis.maximum} (<map input='{axis.maximum}' output='...'>).")
+			if axis.maximum != max(input_locations):
+				raise VarLibValidationError(f"Designspace {ds_path}: Axis '{axis.name}': the axis maximum value must be the highest input value in the mapping.")
+			axis_max_output = axis.map_forward(axis.maximum)
+			if axis_max_output != max(output_locations):
+				raise VarLibValidationError(f"Designspace {ds_path}: Axis '{axis.name}': the axis maximum value must be the highest output value in the mapping.")
+
+			# No duplicate input or output values:
+			input_list = [input for input, _ in axis.map]
+			output_list = [output for _, output in axis.map]
+
+			if len(input_locations) != len(input_list):
+				raise VarLibValidationError(f"Designspace {ds_path}: Axis '{axis.name}': mapping input values must be unique but contain duplicates.")
+			if len(output_locations) != len(output_list):
+				raise VarLibValidationError(f"Designspace {ds_path}: Axis '{axis.name}': mapping output values must be unique but contain duplicates.")
+
+			# Input and output values are in ascending order:
+			if sorted(input_locations) != input_list:
+				raise VarLibValidationError(f"Designspace {ds_path}: Axis '{axis.name}': mapping input values must be in ascending order.")
+			if sorted(output_locations) != output_list:
+				raise VarLibValidationError(f"Designspace {ds_path}: Axis '{axis.name}': mapping output values must be in ascending order.")
```

Comment:
Compared to the previous code, the following mapping which was allowed before is now rejected. Why the change?
```xml
<map input="10" output="1"/>
<map input="0" output="0"/>
<map input="20" output="2"/>
```

Reply:
The specification doesn't seem to impose an order here. @anthrotype sort the map by input values like the previous code so Jany's example passes?

Reply:
yes, sort